### PR TITLE
Split virtual environment detection into a dedicated module

### PR DESCRIPTION
**Pull request by zanieb**
Split out of https://github.com/astral-sh/uv/pull/3266

### CODE REVIEW

File: crates/uv-interpreter/src/environment.rs [hunk 1]
```diff
@@ -152,61 +152,3 @@ impl PythonEnvironment {
         self.interpreter
     }
 }
-
-/// Locate the current virtual environment.
-pub(crate) fn detect_virtual_env() -> Result<Option<PathBuf>, Error> {
```
Comment (zanieb) [inline crates/uv-interpreter/src/environment.rs:157]: Moved into `uv-interpreter::virtualenv`

File: crates/uv-interpreter/src/environment.rs [hunk 2]
```diff
@@ -152,61 +152,3 @@ impl PythonEnvironment {
...
-            debug!("Found a virtualenv named .venv at: {}", dot_venv.display());
-            return Ok(Some(dot_venv));
-        }
-    }
-
-    Ok(None)
-}
-
-/// Returns the path to the `python` executable inside a virtual environment.
-pub(crate) fn detect_python_executable(venv: impl AsRef<Path>) -> PathBuf {
```
Comment (zanieb) [inline crates/uv-interpreter/src/environment.rs:190]: Moved into `uv-interpreter::virtualenv` as `virtualenv_python_executable`

File: crates/uv-interpreter/src/virtualenv.rs [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    ParsePyVenvCfg(PathBuf, #[source] io::Error),
+}
+
+/// Locate the current virtual environment.
+pub(crate) fn detect_virtualenv() -> Result<Option<PathBuf>, Error> {
+    let from_env = virtualenv_from_env();
+    if from_env.is_some() {
+        return Ok(from_env);
+    }
+    virtualenv_from_working_dir()
```
Comment (zanieb) [inline crates/uv-interpreter/src/virtualenv.rs:48]: Split into two methods so we can identify the separate sources in #3266

File: crates/uv-interpreter/src/environment/cfg.rs [hunk 4]
```diff
@@ -1,58 +0,0 @@
-use std::path::Path;
-
-use fs_err as fs;
-use thiserror::Error;
-
-/// A parsed `pyvenv.cfg`
-#[derive(Debug, Clone)]
-pub struct PyVenvConfiguration {
```
Comment (zanieb) [inline crates/uv-interpreter/src/environment/cfg.rs:8]: Moved into `uv-interpreter::virtualenv`

File: crates/uv-interpreter/src/environment/cfg.rs [hunk 5]
```diff
@@ -1,58 +0,0 @@
...
-    /// The version of the `virtualenv` package used to create the virtual environment, if any.
-    pub(crate) virtualenv: bool,
```
Comment (zanieb) [inline crates/uv-interpreter/src/environment/cfg.rs:10]: This docstring is fixed in the moved code

File: crates/uv-interpreter/src/virtualenv.rs [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+};
+
+use fs_err as fs;
+use pypi_types::Scheme;
+use thiserror::Error;
+use tracing::{debug, info};
+
+/// The layout of a virtual environment.
+#[derive(Debug)]
+pub struct VirtualEnvironment {
```
Comment (zanieb) [inline crates/uv-interpreter/src/virtualenv.rs:13]: We might want to make this struct more powerful in the future. Opting not to make more changes now though.

**Review by konstin — APPROVED** [review]

**Pull request merged**